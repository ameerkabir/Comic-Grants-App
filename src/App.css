.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
body {
  /* background-color: #7F85DF; */
  background: #4B51B0;
}

.show-on {
  overflow: auto;
  height: 100%;
  max-width: 75ch;
  cursor: pointer;
}
.show-off {
  overflow: hidden;
  height: 50px;
  max-width: 75ch;
  cursor: pointer;
  
  
}
/* change background when is off */
