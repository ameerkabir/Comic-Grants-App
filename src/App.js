import React, { Component } from "react";
import UserInput from "./UserInput";
import Grants from "./Grants";
import "./App.css";
class App extends Component {
  state = {
    postcode: "",
    lat: "",
    long: "",
    grants: []
  };
  setPostcode = e => {
    this.setState({ postcode: e.target.value });
  };
  search = async e => {
    e.preventDefault();
    try {
      const { postcode } = this.state;
      const { lat, long } = await this.getLongitudeAndLaitude(postcode);
      const response = await this.getAllGrantsByLatLong(lat, long);
      this.setState({ grants: response.data.grants });
    } catch (err) {
      this.setState({
        hasError: true,
        error: "Search did not complete"
      });
    }
  };
  componentDidCatch(error, info) {
    this.setState({
      hasError: true,
      error
    });
    this.logErrorToMyService(error, info);
  }
  logErrorToMyService = err => {
    console.error(err);
  };

  getLongitudeAndLaitude = async postcode => {
    //get info from user postcode
    try {
      const convertUserData = await fetch(
        `https://api.postcodes.io/postcodes/${postcode}`
      );
      const response = await convertUserData.json();
      //Latitude and Longitude from the api response
      let postcodeLat = response.result.latitude;
      let postcodeLong = response.result.longitude;
      return {
        lat: postcodeLat,
        long: postcodeLong
      };
    } catch (err) {
      console.error("Postcode is not valid", err);
    }
  };

  getAllGrantsByLatLong = async (lat, long, range = "15") => {
    try {
      const fetechedUrl = await fetch(
        `https://1kfs7evxca.execute-api.eu-west-1.amazonaws.com/beta/grants-geo/?latitude=${lat}&longitude=${long}&range=${range}km`
      );
      const response = await fetechedUrl.json();
      return response;
    } catch (err) {
      console.log(err);
    }
  };
  render() {
    const { postcode, grants } = this.state;

    return (
      <div className="App">
        <UserInput
          postcode={postcode}
          search={this.search}
          setPostcode={this.setPostcode}
        />
        {this.state.hasError ? (
          <span className="text-danger">
            There was an error performing your search, make sure you have a
            valid postcode.
          </span>
        ) : null}
        <Grants grants={grants} />
      </div>
    );
  }
}

export default App;
