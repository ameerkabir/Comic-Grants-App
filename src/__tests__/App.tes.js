import React from 'react'
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App'

Enzyme.configure({ adapter: new Adapter() });
describe('Component=>App', ()=>{

    it('Should render correctly', ()=>{
        const wrapper = shallow(<App/>)
        expect(wrapper).toMatchSnapshot()
    })
})