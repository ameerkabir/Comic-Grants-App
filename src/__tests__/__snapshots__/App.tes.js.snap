// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component=>App Should render correctly 1`] = `
<div
  className="App"
>
  <UserInput
    postcode=""
    search={[Function]}
    setPostcode={[Function]}
  />
  <Grants
    grants={Array []}
  />
</div>
`;
